pandas/core/generic.py

```diff
@@ -3249,7 +3249,7 @@ def to_csv(
         formatter.save()
 
         if path_or_buf is None:
-            return formatter.path_or_buf.getvalue()
+            return formatter.path_or_buf.getvalue()  # type: ignore
```

Comment:
why is this type ignore? (maybe have to change the return type of to_csv to Optional? or assert this result?)

Reply:
the issue is that `formatter.path_or_buf` can have multiple different types according to the type annotations (not all of them have a `getvalue` method). I will look into whether some `TypeVar` can be used to link the input `file_or_buffer` and the returned` file_or_buffer`.